merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/ConstraintsDSLCompilationService.java

```diff
@@ -0,0 +1,111 @@
+package gov.nasa.jpl.aerie.merlin.server.services;
+
+import gov.nasa.jpl.aerie.json.JsonParser;
+import gov.nasa.jpl.aerie.merlin.server.http.InvalidEntityException;
+import gov.nasa.jpl.aerie.merlin.server.http.InvalidJsonException;
+import gov.nasa.jpl.aerie.merlin.server.models.PlanId;
+import gov.nasa.jpl.aerie.merlin.server.models.ConstraintsCompilationError;
+import gov.nasa.jpl.aerie.merlin.server.models.ConstraintsDSL;
+import org.json.JSONObject;
+
+import javax.json.Json;
+import javax.json.stream.JsonParsingException;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.List;
+import java.util.Objects;
+
+public class ConstraintsDSLCompilationService {
+
+  private final Process nodeProcess;
+  private final TypescriptCodeGenerationService typescriptCodeGenerationService;
+
+  public ConstraintsDSLCompilationService(final TypescriptCodeGenerationService typescriptCodeGenerationService)
+  throws IOException
+  {
+    this.typescriptCodeGenerationService = typescriptCodeGenerationService;
+    final var constraintsDslCompilerRoot = System.getenv("CONSTRAINTS_DSL_COMPILER_ROOT");
+    final var constraintsDslCompilerCommand = System.getenv("CONSTRAINTS_DSL_COMPILER_COMMAND");
+    final var nodePath = System.getenv("NODE_PATH");
+    this.nodeProcess = new ProcessBuilder(nodePath, "--experimental-vm-modules", constraintsDslCompilerCommand)
+        .redirectError(ProcessBuilder.Redirect.INHERIT)
+        .directory(new File(constraintsDslCompilerRoot))
+        .start();
+
+    final var inputStream = this.nodeProcess.outputWriter();
+    inputStream.write("ping\n");
+    inputStream.flush();
+    if (!Objects.equals(this.nodeProcess.inputReader().readLine(), "pong")) {
+      throw new Error("Could not create node subprocess");
+    }
+  }
+
+  public void close() {
+    this.nodeProcess.destroy();
+  }
+
+  /**
+   * NOTE: This method is not re-entrant (assumes only one call to this method is running at any given time)
+   */
+  public ConstraintsDSLCompilationResult compileConstraintsDSL(final PlanId planId, final String constraintTypescript, final String constraintName)
+  {
+    final var missionModelGeneratedCode = JSONObject.quote(this.typescriptCodeGenerationService.generateTypescriptTypesForPlan(planId));
```

Comment:
🔵 Another note for the future: this `JSONObject.quote` calls are the only reason we need the `org.json` dependency. We could refactor this to use `javax.json.JsonObjectBuilder` and remove the dependency.

Reply:
Dropped [AERIE-1815](https://jira.jpl.nasa.gov/browse/AERIE-1815) into the backlog